src/aes_key.c

```diff
@@ -1170,9 +1313,13 @@ int aes_key_sec2prot(struct zpc_aes_key *aes_key, enum aes_key_sec sec)
 		assert(key != NULL);
 		if (is_ep11_aes_key_with_header(key->sec, keylen))
 			return aes_key_sec2prot_with_header(aes_key, sec);
+		else
+			return aes_key_sec2prot_without_header(aes_key, sec);
+	} else if (aes_key->type == ZPC_AES_KEY_TYPE_PVSECRET) {
```

Comment:
I would personally prefer a switch() over aes_key->type here (for better readability), but thats up to you.

Reply:
Done.